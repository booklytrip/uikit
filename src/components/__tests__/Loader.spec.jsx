import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Loader from '../Loader';

describe('Loader', () => {
    it('renders without crashing', () => {
        mount(<Loader />);
    });

    describe('Snapshots', () => {
        it('renders with custom size', () => {
            const rendered = renderer.create(<Loader size="mini" />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders with text', () => {
            const rendered = renderer.create(<Loader text="Loading" />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders inline', () => {
            const rendered = renderer.create(<Loader inline />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
