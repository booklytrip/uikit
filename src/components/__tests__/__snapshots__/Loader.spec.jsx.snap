exports[`Loader Snapshots renders inline 1`] = `
<div
  className="ui active loader medium inline"
  style={undefined} />
`;

exports[`Loader Snapshots renders with custom size 1`] = `
<div
  className="ui active loader mini"
  style={undefined} />
`;

exports[`Loader Snapshots renders with text 1`] = `
<div
  className="ui active loader medium text"
  style={undefined}>
  Loading
</div>
`;
